optimum/neuron/pipelines/diffusers/pipeline_stable_diffusion_xl.py

```diff
@@ -16,11 +16,20 @@
 from typing import Any, Callable, Dict, List, Optional, Tuple, Union
 
 import torch
-from diffusers import StableDiffusionXLPipeline
-from diffusers.loaders import LoraLoaderMixin, TextualInversionLoaderMixin
-from diffusers.pipelines.stable_diffusion_xl import StableDiffusionXLPipelineOutput
-from diffusers.pipelines.stable_diffusion_xl.pipeline_stable_diffusion_xl import rescale_noise_cfg
-from diffusers.utils import randn_tensor
+
+from ....utils import is_diffusers_available
+
+
+if is_diffusers_available():
+    from diffusers import StableDiffusionXLPipeline
+    from diffusers.loaders import LoraLoaderMixin, TextualInversionLoaderMixin
+    from diffusers.pipelines.stable_diffusion_xl import StableDiffusionXLPipelineOutput
+    from diffusers.pipelines.stable_diffusion_xl.pipeline_stable_diffusion_xl import rescale_noise_cfg
+    from diffusers.utils.torch_utils import randn_tensor
+else:
+    raise ImportError(
```

Comment:
here @michaelbenayoun

Reply:
No we dont want that right? Otherwise reading this file will end-up making everyone fail. Or we should make sure that it is never imported anywhere where `diffusers` is not checked for explicitly.

Reply:
How it end up making __everyone__ fail? It doesn't solve the training test issue though. I will remove changes in pipeline files.
These pipelines are only imported under `modeling_diffusion.py` , will put the check there.

Reply:
Alright!